src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -325,25 +325,25 @@ public void FullValidationSucceeded(ChainedHeader chainedHeader)
         }
 
         /// <inheritdoc />
-        public List<ChainedHeaderBlock> PartialValidationSucceeded(ChainedHeader chainedHeader, out bool fullValidationRequired)
+        public PartialValidationSucceededResult PartialValidationSucceeded(ChainedHeader chainedHeader)
```

Comment:
Perhaps this is a good change but it could be done in a pure env and not as part of an important fix

Reply:
I don't understand why, if improvements are introduced which is only going to help us debug the problem faster, they are frowned upon and asked not to be included in a PR. To create a new PR and go through the whole build and testing process will just waste a lot of time.

I really don't agree with this way of writing code.

If you can honestly tell me that the new code is worse then sure, no problem but just dismissing code because its slightly more difficult to review is not good enough reason.

Reply:
i disagree with your disagreement. this is critical fix, it  should be as small as possible, not touching ANYTHING else, no agent prefix, no block puller, no CHT, NOTHING. this only makes it harder for everyone to review this and i have asked multiple times not to do this exact thing to include unrelated code in PR, especially if it is bug fixing PR. yet you ignored it again :( that is why i feel i need to block it until everything unrelated is out of it

Reply:
@Aprogiena I disagree, any changes realting to the bug and slight improvements should be regarded and reviewed on merit. 

Granted, the other fixes to BlockPuller shouldn't have been in here as well as the "supposed" fix for the `UnconsumedBytes` null reference, although if I was the reviewer here I wouldn't have had an issue with this. A simple message to me asking why the change would've explained it.

Changes to the tests were there for me to better understand which nodes are syncing and when, hence the agent prefix improvement. Again, a simple "why was this changed" would have sufficed.

Apart from the BlockPuller change, ALL of these relates to this ONE issue @Aprogiena and as such they are good and relevant.

Reply:
I did ask you that question and you replied and you said

```
I had to refactor OnPartialValidationSucceededAsync so that we work with a PartialValidationSucceededResult to determine what should happen next. 
```

which is wrong because that was not necessary (as my suggestion of fix proves). Therefore your approach does not stand. Also you make arbitrary new allocation which is not needed and then you have a hidden functionality based on internal property being null or just empty, which is quite smelly approach. Then you violate logging policies, but that's just tiny little thing at the end. 

Changes in tests that are not required for the new test you implemented don't belong here as well. 

So the whole change is based on wrong reasoning based on you not understanding the code.  That is not a reason that has merit and does not justify the change.

So I repeat, this is important bugfix, it should be minimal, implemented with at most 3 line change + comment + test. Nothing else.

You can't argue that any change that is good belongs everywhere. That's total nonsense, because you would suddenly combine everything everywhere. This is such a waste of time and I am asking you again - please don't do it.

Reply:
- which is wrong because that was not necessary (as my suggestion of fix proves).

Why is returning a result class wrong? Just because you say so? Working with result class is more intuitive than calling a Validate method and returning a set of headers to validate. In no world is this something a new dev would understand as normally calling a validate method should return a validation result (and not some `out` parameter). We are working with C# here you know, we are allowed to make cahnges to make our life easier.

- hidden functionality based on internal property being null or just empty, which is quite smelly approach

What hidden functionality? You say this is a waste of time but again (and just like the last time you just dismissed my PR) you are just commenting without any constructive criticism or actually telling me what is wrong with the code. This is very unprofessional and not helping the cause. 

- Then you violate logging policies, but that's just tiny little thing at the end.

Again, why cant you just tell me what was being violated?

- So the whole change is based on wrong reasoning based on you not understanding the code. That is not a reason that has merit and does not justify the change.

This is **exactly** a reason to change the code, I am an experience developer and have been here just as long as you and if even I cant understand the code then surely a new dev won't so why not improve on it a bit here and now?

- You can't argue that any change that is good belongs everywhere. That's total nonsense, because you would suddenly combine everything everywhere. 

I didn't say that, the changes I made was for the betterment of the code and for me (and future devs) to understand what was done better.

It pains me that we are in a place where code cant be reviewed on merit. If you actually spent 5-10minutes reviewing the code (3 hours ago) and making constructive comments on them we would have had this PR merged by now and everyone would have been able to move along.

You can block this, but you are now holding up everyone with this.

Reply:
1) three people are telling you that simple fix is better
2) your code is confusing (https://github.com/stratisproject/StratisBitcoinFullNode/pull/2216#discussion_r218489274)
3) there exists trivial solution that fixes the problem

just this should be enough, if it is not, then i can't do anything, we have simply too different approaches and are not compatible